### PR TITLE
add support for PSBT signing endpoint

### PR DESCRIPTION
**Pull request by kwantam**
This PR adds support for the new PSBT signing endpoint.

It also updates yaml.v3 to address a dependabot complaint.

### CODE REVIEW

File: api/v0/psbt_sign_200_response.go
Comment (kwantam) [inline api/v0/psbt_sign_200_response.go:1]: openapi generator

File: api/v0/psbt_sign_request.go
Comment (kwantam) [inline api/v0/psbt_sign_request.go:1]: openapi generator

File: api/v0/psbt_sign_response.go
Comment (kwantam) [inline api/v0/psbt_sign_response.go:1]: openapi generator

File: client/btc.go [hunk 1]
```diff
@@ -107,3 +107,52 @@ func (cli *Client) SignSegWit(roleId, pubkey string, request *v0.BtcSignRequest,
 	}
 	return &decoded, "", nil
 }
+
+func (cli *Client) SignPsbt(roleId, pubkey string, request *v0.PsbtSignRequest, mfaId, mfaConfirmation *string) (*v0.PsbtSign200Response, string, error) {
```
Comment (kwantam) [inline client/btc.go:111]: essentially copypasta from function above

File: client/btc.go [hunk 2]
```diff
@@ -107,3 +107,52 @@ func (cli *Client) SignSegWit(roleId, pubkey string, request *v0.BtcSignRequest,
...
+	authResp, err := cli.CreateRoleToken(&v0.CreateTokenRequest{
+		Purpose: "sign psbt",
+		Scopes:  []api.Scope{api.SIGNBTCPSBT},
```
Comment (deian) [inline client/btc.go:116]: - ack that you're following the convention above
- this and other bits should probably be refactored to not need management scopes

**Review by deian — APPROVED** [review]

**Pull request merged**